File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
        public static int solution(String s) {
        int answer = 0;

        for (int i = 0; i < s.length() - 1; i++) {
            Stack<Character> stack = new Stack<Character>();
            String new_s = s.substring(i) + s.substring(0, i);
            int cnt = 0;
            char c = 'c';
            for (char ch : new_s.toCharArray()) {
                if (ch == '(' || ch == '{' || ch == '[') {
                    stack.add(ch);
                }

                if ((!stack.empty() && stack.peek() == '(' && ch == ')')
                        || (!stack.empty() && stack.peek() == '[' && ch == ']') ||
                        (!stack.empty() && stack.peek() == '{' && ch == '}')) {
                    stack.pop();
                    cnt += 1;
                }
            }
            if (stack.empty() && cnt > 0) {
                answer += 1;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    for i in range(-1,len(s)-1):
        tmp = s[i+1:]+s[:i+1]
        stack = []
        cnt = 0
        for t in tmp:
            # print(stack)
            if t == "[" or t == "{" or t == "(":
                stack.append(t)
                cnt +=1

            if stack and stack[-1] == "[" and t == "]":
                stack.pop()
                cnt +=1
            elif stack and stack[-1] == "{" and t == "}":
                stack.pop()
                cnt +=1
            elif stack and stack[-1] == "(" and t == ")":
                stack.pop()
                cnt +=1

        if not stack and cnt > 0:
            answer +=1

    return answer